meson.build

```diff
@@ -129,170 +129,207 @@ thread_dep = dependency('threads') # pthread for tensorflow-lite
+#      following variables handled with _when_available/_when_not_available postfix,
+#      project_args: <dict>, message: <string>, extra_args: <dict: meson variables to be registered>
+# } ]
+
+features = {
+  'video-support': {
+    'is_available': not get_option('video-support').disabled(),
+    'project_args_when_not_available': { 'NO_VIDEO': 1 },
+  },
+  'audio-support': {
+    'is_available': not get_option('audio-support').disabled(),
+    'project_args_when_not_available': { 'NO_AUDIO': 1 },
+  },
+  'tf-support': {
+    'deps': [ 
+      dependency('tensorflow', required: get_option('tf-support')), 
+      protobuf_dep
+    ],
+    'project_args_when_available': { 'ENABLE_TENSORFLOW': 1 },
+  },
+  'tflite-support': {
+    'deps': [ dependency('tensorflow-lite', required: get_option('tflite-support')) ],
+    'project_args_when_available': { 'ENABLE_TENSORFLOW_LITE': 1 }
+  },
+  'pytorch-support': {
+    'deps': [ dependency('pytorch', required: get_option('pytorch-support')) ],
+    'project_args_when_available': { 'ENABLE_PYTORCH': 1 }
+  },
+  'caffe2-support': {
+    'deps': [ dependency('pytorch', required: get_option('caffe2-support')) ],
+    'project_args_when_available': { 'ENABLE_CAFFE2': 1 }
+  },
+  'mvncsdk2-support': {
+    'deps': [ cc.find_library('mvnc', required: get_option('mvncsdk2-support')) ],
+    'is_available_fallback': cc.check_header('mvnc2/mvnc.h', required: get_option('mvncsdk2-support') ),
+    'project_args_when_available': { 'ENABLE_MOVIDIUS_NCSDK2' : 1}
+  },
+  'nnfw-runtime-support': {
+    'deps': [ 
+      nnfw_dep,
+      dependency('json-glib-1.0', required: get_option('nnfw-runtime-support')) 
+    ],
+    'project_args_when_available': { 'ENABLE_NNFW_RUNTIME': 1 }
+  },
+  'tflite-nnapi-delegation': {
+    'deps': [ dependency('nnfw', required: get_option('tflite-nnapi-delegation')) ],
+    'project_args_when_available': { 'ENABLE_TFLITE_NNAPI_DELEGATION': 1 }
+  },
+  'armnn-support': {
+    'deps': [ dependency('armnn', required: get_option('armnn-support')) ],
+    'project_args_when_available': { 'ENABLE_ARMNN': 1 }
+  },
+  'orcc-support': {
+    'deps': [ orc_dep , pg_orcc ],
+    'project_args_when_available': {'HAVE_ORC': 1},
+    'project_args_when_not_available': { 'DISABLE_ORC': 1 },
+    'extra_args_when_available': {'orcc_args': [pg_orcc, '--include', 'glib.h'] }
+  }
+}
+
+project_args = {}
+# This section controls the flow of feature registration.
+foreach feature_name, data : features
+  _deps = data.get('deps', [])
+  _deps_all_found = true
+
+  foreach dep : _deps
+    if not dep.found()
+      _deps_all_found = false
+    endif
+  endforeach
 
-  if tf_dep.found() and protobuf_dep.found()
-    add_project_arguments('-DENABLE_TENSORFLOW=1', language: ['c', 'cpp'])
-  else
-    error('Cannot find tensorflow')
+  if not _deps_all_found
+    warning('Target dependency is not found for feature @0@'.format(feature_name))
```

Comment:
If dependency it not met, it's an error unless user has specified it is "optional". (Let's not handle "optional" dependencies in this automated script for now; we are not using it here, yet.)

Reply:
Or.. for fallback (or alternative), if the alternative dep check exists, do it here and emit error if that fails too. (similar semantics in similar lines)

Reply:
I changed not to put any fallback logic for now. 
Fallback has many cases making logic more complex.
eg. `deps_1.found() or deps_2.found()` , `dep1_header_found and deps_2.found() or deps_3.found()`, etc...

So, I revised to handle fallback logic separately for more simplicity.

It will fail at `dependency('...', required: get_option(<feature>))` if it has to be failed.